solr/core/src/java/org/apache/solr/search/neural/KnnQParser.java

```diff
@@ -72,7 +81,49 @@ public Query parse() {
 
     DenseVectorField denseVectorType = (DenseVectorField) fieldType;
     float[] parsedVectorToSearch = parseVector(vectorToSearch, denseVectorType.getDimension());
-    return denseVectorType.getKnnVectorQuery(schemaField.getName(), parsedVectorToSearch, topK);
+
+    return getFilterQuery()
+        .map(
+            filterQuery ->
+                denseVectorType.getKnnVectorQuery(
+                    schemaField.getName(), parsedVectorToSearch, topK, filterQuery))
+        .orElse(
+            denseVectorType.getKnnVectorQuery(schemaField.getName(), parsedVectorToSearch, topK));
+  }
+
+  private Optional<? extends Query> getFilterQuery() throws SolrException {
+    if (!isFilter()) {
+      String[] filterQueries = req.getParams().getParams(CommonParams.FQ);
+      return ofNullable(filterQueries).filter(fq -> fq.length != 0).map(this::computeFilterQuery);
+    }
+    return Optional.empty();
+  }
+
+  private Query computeFilterQuery(String[] filterQueries) throws SolrException {
+
+    List<Query> filters = new ArrayList<>(filterQueries.length);
+
+    for (String filterQuery : filterQueries) {
+      if (filterQuery != null && filterQuery.trim().length() != 0) {
+        QParser filterParser;
+        try {
+          filterParser = QParser.getParser(filterQuery, super.getReq());
+          filterParser.setIsFilter(true);
+          filters.add(filterParser.getQuery());
+        } catch (SyntaxError e) {
+          throw new SolrException(
+              SolrException.ErrorCode.BAD_REQUEST,
+              "error in parsing filter queries:" + e.getMessage());
+        }
+      }
+    }
+
+    BooleanQuery.Builder builder = new BooleanQuery.Builder();
+    for (Query query : filters) {
+      builder.add(query, BooleanClause.Occur.FILTER);
+    }
+
+    return builder.build();
```

Comment:
If `filters` contained only one element, could that be returned directly?

Reply:
Good point. I changed the code in order to return it direcly